ext/nokogiri/xml_text.c

```diff
@@ -20,7 +20,7 @@ new (int argc, VALUE *argv, VALUE klass)
 
   rb_scan_args(argc, argv, "2*", &string, &document, &rest);
 
-  Data_Get_Struct(document, xmlDoc, doc);
+  doc = (xmlDocPtr)noko_xml_node_unwrap(document);
```

Comment:
`(xmlDocPtr)noko_xml_node_unwrap(document)` happens often enough that I think we should have a `noko_xml_document_unwrap()` although I don't feel super strongly about it.

Reply:
Yeah, that jumped out at me, too. I may try some variations on this this weekend. I may also try using a macro with the same parameters as `Data_Get_Struct`.